### PR TITLE
Obsolete several monsters: poor references & non-native fauna

### PR DESCRIPTION
**Pull request by Karol1223**
#### Summary
None

#### Purpose of change
Obsoletes several monsters that were either generally regarded as unfitting or are animals not native to the region the game takes place in

#### Describe the solution
**CYCLOPEAN**
Pretty poor reference to Futurama. For some reason spawns randomly around basements. Has the weirdest loot table on this side of the universe. On the devcord always brought up as a poor unfitting joke.

**RAT KING**
Yes, they do exist irl. No, they don't exist anywhere near what we had. A weird completely unexplainable mass of fused rats that mutates you with insults. I admire the creativity but no, I cannot pretend like it fits anywhere in the mainline. Having said that, I lowkey feel like it *may* fit Xedra so Xedra Maintainers - lemme know if you want to adopt it. **EDIT: IT HAS BEEN MOVED TO XEDRA**

**GIGANTIC NAKED MOLERAT**
Molerats are not native to New England and are really high maintenance to keep alive so they make poor test animals. Plus I have a hunch it's just a nod to Fallout. It's also not really implemented properly - no size variation in a mutant, they spawn equivalent to mega mutant insects or the deathrattle serpent which are the FINAL forms of those mutants. Yes, it could be reflavored to be a regular mutant mole. But the problem of no size forms still stands and... normal moles are ungodly boring. If you want to remake it properly as a normal mole be my guest but it'd have to be an entirely new monster either way.

**REINDEER, REINDEER FAWNS AND ZEINDEER**
Perfect animal - imperfect idea. Reindeers are not native to New England. Next, please. **EDIT: IT HAS BEEN RESTORED AND MADE A CHRISTMAS EVENT SPAWN. SPAWNS IN MEGAFAUNA ALL YEAR ROUND**

**BLINKY**
Old reference to the Simpsons. Is a mutant but also gives normal meat. For some reason it weighs 40kg???? Anyway, yeah, a relic of days passed. Not a fan. Yote.

**KILLER BUNNY**
Removed alongside its mapgen (although I kept the blood cave in Innawood, it's cool and I hope someone will shit themselves). It's a Monty Python reference that came with a free mininuke. Please do not the bnuuy.

#### Describe alternatives you've considered
**MASS OF ZOMBIE SPIDERS**
Entirely unfitting to the lore but Venera has been bullying me about it being a cool concept so it should be reworked

**CRAWLER**
Who the fuck knows what that is? But someone on devcord wanted to rework it so it stays

**JAWED TERROR**
I hate these fucks and I'm coming for them. But obsoleting them right now would leave the Prison Island start thoroughly broken. So they stay until I find a replacement.

### CODE REVIEW

**Comment by Maleclypse** [comment]
What about moving reindeer / zeindeer to the seasonal monster list? I’ll take the rat king in Xedra Evolved.

**Comment by Karol1223** [comment]
> What about moving reindeer / zeindeer to the seasonal monster list?

You mean having them only spawn during christmas? It's... an interesting idea, to be honest. 
Having said that, I have no idea how to do that. Do we have examples of this in the code?

> I’ll take the rat king in Xedra Evolved.

I'll throw it in there in a bit and copy the rat cave mapgen with it as well

**Comment by GuardianDll** [comment]
*cough cough* it was originally intended to be seasonal, in #45675

**Comment by Maleclypse** [comment]
https://github.com/CleverRaven/Cataclysm-DDA/pull/53100

**Comment by Karol1223** [comment]
Gotcha, I'll handle this tomorrow then alongside moving the Rat King to Xedra. Thanks for the examples!

**Comment by Theawesomeboophis** [comment]
Most of these could probably be put in Crazy, I planned on changing reindeer to be event spawns when the update came out but I forgot about it as I'm focused on other things.

**Comment by irwiss** [comment]
May be delete the monster json entirely, and delete the existing monster obsoletion file
After https://github.com/CleverRaven/Cataclysm-DDA/pull/64496 monsters with non-existing type load as factionless breathers

There's not much benefit in "carrying" the obsolete monster json until next stable - as the "migration" process for them in next stable is likely to be the same as it is now

**Comment by 7erracotta** [comment]
> **CRAWLER** Who the fuck knows what that is? But someone on devcord wanted to rework it so it stays

I think people not knowing what it is, is what makes really cool,  not every creepy monster needs an explanation for its existence. (I hope this doesn't sound mean or rude!)

Also, I'm VERY surprised Mausketeer isn't on here, as that enemy is just a straight-up rat in clothes wielding a sword.

**Comment by actually-the-artificer** [comment]
> > **CRAWLER** Who the fuck knows what that is? But someone on devcord wanted to rework it so it stays
> 
> I think people not knowing what it is, is what makes really cool, not every creepy monster needs an explanation for its existence. (I hope this doesn't sound mean or rude!)
> 
> Also, I'm VERY surprised Mausketeer isn't on here, as that enemy is just a straight-up rat in clothes wielding a sword.

It only spawns in a special location iirc, and has an explanation for how it came to be.

**Comment by MNG-cataclysm** [comment]
The Giant Molerats should probably be changed into a proper giant mole, one native to New England. The monster is currently used as part of a mission.

**Comment by Karol1223** [comment]
> May be delete the monster json entirely, and delete the existing monster obsoletion file

On it boss

> Most of these could probably be put in Crazy

Crazy Cata really should be funny and not a dumping ground. The only real contender for it is the Killer Bunny which I'm coming for next

> not every creepy monster needs an explanation for its existence.

Not every monster needs an explanation but every one needs a place in the lore

> Also, I'm VERY surprised Mausketeer isn't on here, as that enemy is just a straight-up rat in clothes wielding a sword.

I dislike the flavor of the Mausketeer but it does have a lore explanation so it doesn't go with the fellas I have here

> The Giant Molerats should probably be changed into a proper giant mole, one native to New England. The monster is currently used as part of a mission.

In the PR description I explained why I will not do that. The mission is also not a concern to me at all because it's a mission for an obsoleted faction so it's not my problem.

**Comment by merrygin** [comment]
I would like to move the reindeer and reindeer fawn to Megafauna. Apparently, they used to be native to northern New England up until the 70s but were driven out due to habitat loss/overpredation.

**Comment by PatrikLundell** [comment]
The killer worms make no sense either (there's a refugee center quest for one, making removal an issue, though).

**Comment by Karol1223** [comment]
> The killer worms make no sense either (there's a refugee center quest for one, making removal an issue, though).

What doesn't make sense about them is their aggressive nature, to be honest. Mutant earthworms in themselves fit perfectly.

**Comment by Zireael07** [comment]
I'd be in favor of improving the molerats instead of axing them - they're a fitting spawn in sewers/caves and removing them would greatly reduce the danger in there (which tbh is too low imho)

**Comment by Karol1223** [comment]
> I'd be in favor of improving the molerats instead of axing them - they're a fitting spawn in sewers/caves and removing them would greatly reduce the danger in there (which tbh is too low imho)

I fully support the remake of their concept into something more fitting but like I said - that'd need to be an entirely different creature. Regular moles are too boring. People on the devcord suggested a nether tunneling monster which is good but I will not put time and energy into it. They'll be cut and when someone wants to reuse their concept for a different critter they can Frankenstein parts of their dead code into it.

**Comment by fairyarmadillo** [comment]
The mass of zombie spiders is a weird lore conflict and should get nixed. It also doesn't really behave like a swarm of things, it's just a single monster, kinda boring.

**Comment by Karol1223** [comment]
> The mass of zombie spiders is a weird lore conflict and should get nixed. It also doesn't really behave like a swarm of things, it's just a single monster, kinda boring.

The rework idea Venera keeps pestering me with is some form of an artifact that'd control non-mutated spiders to swarm around it but
- I really hate nether and artifacts
- I can't give that idea justice if I wanted to

So I'm not gonna bother. But I'm not removing it because there was a will of someone offering to rework it so for now it stays. Whereas with the Naked Molerats for example nobody offered shit, it was just throwing around ideas.

Words mean nothing. Actions do.

**Comment by Karol1223** [comment]
Okay so updates:
- Rat King and its old cave mapgen got moved to XEDRA Evolved
- Reindeer are back and made a Christmas seasonal spawn. Megafauna has them spawning all year round
- Killer Rabbit is gone alongside its mapgen. Although I kept the bloody cave for Innawoods because I like the mental image of players shitting themselves when they find it and its empty

**Comment by Karol1223** [comment]
> Does that mean Venera's going to rework it?

Ha. Likely not, though I do keep pretending like I'll get him to do that one day. You're right in that they should probably be obsoleted but they stay for the same reason I didn't touch Gas zombies - there's a concrete concept for its rework. Meanwhile, Naked Molerats were just people spitballing random ideas.

I will not oppose the obsoletion of the mass of zombie spiders but I don't feel comfortable removing it alongside these pals I've got here.

**Comment by SkitariiAuxiliary** [comment]
And this is why we cannot have nice things. You don't like something so no one else is allowed to have it either.

**Comment by Karol1223** [comment]
> And this is why we cannot have nice things. You don't like something so no one else is allowed to have it either.

Exactly! Me not liking something is the single reason why it doesn't fit the lore. You got it 100% right.

**Comment by Zireael07** [comment]
@Karol1223 Just a word of warning - sarcasm doesn't convey well on the internet

**Comment by Karol1223** [comment]
> @Karol1223 Just a word of warning - sarcasm doesn't convey well on the internet

It doesn't have to, I'll let the people with merge rights sort this stuff out. Thumbs down don't block a PR from being merged, luckily, otherwise we'd never get any nerfs merged.

**Comment by toweroglass** [comment]
i think deleting the molerat prior to a rework seems a little silly. 

if reindeers are being nixed they should be added as spawns in the zoo. Non native animals make a ton of sense in a zoo and dumping fleshed out mobs haphazardly doesn't seem fun. 
Wouldn't be a terrible idea to bring over some animals from tropicatalysm for the zoo as well.

**Comment by Fris0uman** [comment]
Make rat king a nether monster that spawns some shadow rats ( like the shadow snake thing in the cabin) instead maybe. Some recent change to the refugee center arsonist imply that portal storm can induce mutation so you might not even need to remove the mutation power from the rat king

**Comment by Karol1223** [comment]
> Some recent change to the refugee center arsonist imply that portal storm can induce mutation so you might not even need to remove the mutation power from the rat king

Oh nether definitely has the power to mutate - I mean we've got Yuggs for starters!
Though personally I think Rat King is better off as a non-reworked monster for Xedra than a forced rework for Mainline. I'd also rather really not touch nether content. It's far from my cup of tea and I know for a fact I'd only do the bare minimum to get it merged, it wouldn't be quality content.

**Comment by John-Candlebury** [comment]
I dont like the removal of the rat king either, I see nothing weird in the way it exists now, even if it could feel more supernatural.

**Comment by Karol1223** [comment]
> I dont like the removal of the rat king either, I see nothing weird in the way it exists now, even if it could feel more supernatural.

It's an urban legend come-alive with no clear connection to the Nether and that mutates the player by calling them filth. It also feels like something straight out of an RPG. So I completely disagree.

**Comment by fairyarmadillo** [comment]
Nether really often just feels like a fall-back because there's not really a lot of cohesive lore for it. Weird ghosts, alien fungi, psychic attacks and dimensional effects have some thematic consistency, even though it's thin. A mammal from earth that speaks english and, for some reason, is able to perfectly replicate the effects of a very specific mutagen developed in a laboratory by talking is comparitively nonsensical.

**Comment by merrygin** [comment]
Re:putting some of these into Crazy Cataclysm, possibly all the reference-style monsters have a place there (Blinky, Cyclopean, Killer Bunny)? I don't know exactly the design philosophy behind Crazy, but these seem like conceptually funny easter eggs. 

Also, as many here do seem to really like those types of monsters with limited realism/too much of a lore-conflict/easter-eggs, perhaps an alternative would be to actually make a "dumping ground" mod for weird and zany monsters like these that's not a full overhaul like Crazy. Of course some additional work, but if a lot of people would like these to stay, I do think it'd be important to take that into account/take it seriously. :)

**Comment by Karol1223** [comment]
> Re:putting some of these into Crazy Cataclysm, possibly all the reference-style monsters have a place there (Blinky, Cyclopean, Killer Bunny)? I don't know exactly the design philosophy behind Crazy, but these seem like conceptually funny easter eggs.

You kinda hit the spot. Conceptually funny. Out of those Killer Rabbit is the only real contender (which I either mentioned here on discord that I did consider porting it over)

> Also, as many here do seem to really like those types of monsters with limited realism/too much of a lore-conflict/easter-eggs, perhaps an alternative would be to actually make a "dumping ground" mod for weird and zany monsters like these that's not a full overhaul like Crazy.

We have a history of dumping ground mods that nobody in the development, active anyway, seems to have any positive feelings towards and neither do I. I will not tell you not to attempt to make one but also don't expect me to do that and I can't guarantee that'd be merged.

> a lot of people would like these to stay, I do think it'd be important to take that into account/take it seriously. :)

There are more important factors to consider here. For starters, Blinky is Disney property that's not even renamed for god's sake. Foolproofing it shouldn't be hard but this is just an example. Second of all - what people want is of secondary importance, as evidenced by the dozens of harsh obsoletion PRs before this one. I won't disagree fully with you - people's opinions should be taken into account when viable, but I will not go the extra mile to cater to the community. Any of you are welcome to do so, anyone can make a PR.

File: data/json/monstergroups/wilderness.json
Comment (LyleSY) [inline data/json/monstergroups/wilderness.json:225]:
```suggestion
```
Please remove this line, it's duplicative in intent with what you're doing and what you're doing is better

File: data/json/npcs/valhalla_cult/missions.json
Comment (LyleSY) [inline data/json/npcs/valhalla_cult/missions.json:11]:
```suggestion
      "assign_mission_target": { "om_terrain": "field", "reveal_radius": 1, "random": true, "search_range": 20, "min_distance": 5, "z": 0 },
      "update_mapgen": { "place_monster": [ { "monster": "mon_wolf_mutant_huge", "x": 11, "y": 11, "target": true } ] }

```
Suggest replacing monster to not break this mission chain. A quick google suggests big wolves are very important in Norse mythology, more so than giant naked rats anyway

**Review by LyleSY — COMMENTED** [review]
Overall makes sense. I recommend tweaking the zeindeer spawns slightly and changing the giant rat quest to be a giant wolf quest instead of breaking it. Please also change the "rat" references in that mission to be "wolf". Otherwise good stuff here, thanks for being the villain.

**Comment by Karol1223** [comment]
> Overall makes sense. I recommend tweaking the zeindeer spawns slightly and changing the giant rat quest to be a giant wolf quest instead of breaking it. Please also change the "rat" references in that mission to be "wolf". Otherwise good stuff here, thanks for being the villain.

I'll tackle this right away. And you are rightt with the wolf - hell, we even have mutant wolves which are ready for this. Initially I couldn't care less about the mission but now that the vikings may just be coming back I guess I have to take this into account seriously.

**Review by LyleSY — APPROVED** [review]

**Comment by LyleSY** [comment]
What's your concern with jawed terrors and can I interest you in a delicious zombie catfish?

**Comment by Karol1223** [comment]
> What's your concern with jawed terrors and can I interest you in a delicious zombie catfish?

They're not native to the region. Bull sharks are extremely rare that far north and they only go into freshwater to breed and give birth so they wouldn't be in New England waterways.

I intend to replace them once I am done with my fish rework. The idea is to first add some more species, then rework current fish mutants and then add more fish mutants. I want to replace jawed terrors with mutant pikes.

**Comment by MNG-cataclysm** [comment]
> I intend to replace them once I am done with my fish rework. The idea is to first add some more species, then rework current fish mutants and then add more fish mutants. I want to replace jawed terrors with mutant pikes.

What about giant, murderous carp that stands up?

File: data/mods/Xedra_Evolved/mapgen/cave.json
Comment (bombasticSlacks) [inline data/mods/Xedra_Evolved/mapgen/cave.json:80]: There doesn't appear to be a rat king in your rat king cave

Comment (Karol1223) [reply data/mods/Xedra_Evolved/mapgen/cave.json:80]: Ah yeah, it would appear someone copied over the wrong json. Give me a second

**Comment by Theawesomeboophis** [comment]
Why not make Jawed Terrors only spawn in summer as that's when they usually migrate, around 250 sharks were reported in the New England area in 2020. Source: https://www.nbcboston.com/news/local/heres-where-sharks-were-spotted-in-new-england-this-past-week/2170975/

**Comment by Karol1223** [comment]
> Why not make Jawed Terrors only spawn in summer as that's when they usually migrate, around 250 sharks were reported in the New England area in 2020

The simple answer - as I said, they only go to freshwater to breed. Bull sharks don't breed that far north because it's too cold for them. The ones that do go into freshwater in colder climates are confused stragglers. They have no business being in the game until we have the ocean.

**Comment by Theawesomeboophis** [comment]
Bull sharks aren't the only type of shark that migrates to New England though.

**Comment by Theawesomeboophis** [comment]
Fair point about them being the only freshwater sharks. Not trying to argue btw, was just wondering why they were being obsoed instead of an alternative solution. I understand your reasoning now, sorry if it seemed argumentative, I sincerely didn't intend for it to be.

**Comment by Karol1223** [comment]
No, I'm sorry, I'm just extremely tired of this topic. On discord I've had the same person continuously ask me why I don't want Jawed Terrors in the game so I had to retrace this exact conversation 5 times for *one person* and I'm just really tired.

I do want an alternate solution - mutants of actual freshwater fish, that's what I'm going to go with by the time I do indeed touch Jawed Terrors.

**Comment by Theawesomeboophis** [comment]
That sounds cool, I would love an expansion of aquatic mutants. Can't wait to see what you have planned.

**Review by MNG-cataclysm — APPROVED** [review]
Everything looks good. You've done a great job with this.

**Comment by sonphantrung** [comment]
So the 'Island Prison' scenario will become less intense?

**Comment by Karol1223** [comment]
Why do people keep talking like I've obsoleted Jawed Terrors here. I didn't. I plan to do that in the future.

No, I hope it doesn't. We'll see how it plays out when I get there though.

**Comment by bombasticSlacks** [comment]
Alright, so I'm going to merge this,

The following is just my opinion but:

If someone wanted to improve the quality of some of the jokes/references perhaps they would fit in Crazy Cata, the issue is that mod doesn't have much maintenance.

I think mole rats need to be replaced with something less disruptive, they are terrible for destroying locations while in the reality bubble, something an equivalent creature would have not to do at a similar rate. This goes back to the classic "what does this look like if the reality bubble doesn't exist" and if the answer is that every mole rat location becomes a trash pile, that's not great.

I also think the rat king is cool; however, it's a bit of a gotcha death trap right now. I think if someone wanted to give it some love and migrate it to a more appropriate location like the sewers, subway tunnels, or perhaps TCL maintenance with its own clearly demarcated arena/area would be ideal. Potentially some rewards could be brought back in that case too since it would then be in proximity to advanced items. The JSON for this one is also kept in XEDRA, so it's not like it's getting lost.

**Comment by Karol1223** [comment]
> I also think the rat king is cool; however, it's a bit of a gotcha death trap right now. I think if someone wanted to give it some love and migrate it to a more appropriate location like the sewers, subway tunnels, or perhaps TCL maintenance with its own clearly demarcated arena/area would be ideal. Potentially some rewards could be brought back in that case too since it would then be in proximity to advanced items. The JSON for this one is also kept in XEDRA, so it's not like it's getting lost.

Just to add some last thoughts and explain myself here

For the record, I actually loved the Rat King. It was the only monster in this PR that I genuinely liked and thought was implemented well. But the feel of it doesn't sit right.

As it stands Rat King is a discount Yugg aimed at one specific mutation. It has no clear tie-ins to the nether, the ratkin, the mutants, *anything*. It just has the ability to mutate you with words. Which is cool on paper but not cool when it's not explained in any way and it doesn't tie to any other existing content.

To fit the basegame well Rat King would need plenty of TLC and not only can I not provide it - I full-heartedly believe that giving it a rework would destroy the charm it has now. I am of 100% belief that it is better for our loved and well-functioning, well-flavored Rat King to stay as is and at home in Xedra than to rework it into something that'll be subpar to the original.

Obviously, I have no power to stop anyone from reworking it and bringing it back, but that's just my two cents here.

**Comment by Proxima-b** [comment]
So how many kinds of monsters do we still have in innawood ?
What can I fight with my late-game muskets ?

**Comment by FayFlightee** [comment]
> I also think the rat king is cool; however, it's a bit of a gotcha death trap right now. I think if someone wanted to give it some love and migrate it to a more appropriate location like the sewers, subway tunnels, or perhaps TCL maintenance with its own clearly demarcated arena/area would be ideal. Potentially some rewards could be brought back in that case too since it would then be in proximity to advanced items. The JSON for this one is also kept in XEDRA, so it's not like it's getting lost.

That does sound good, giving it a proper area instead.
...when I figure out how to make map area, I'll probably do it. Though, I'm in no place to explain it lore-wise.

**Pull request merged**
